jobs/mongodb_backup_job.go

```diff
@@ -147,19 +186,95 @@ func createDBURL(dbConfig DBConnConfig) *url.URL {
 	}
 }
 
-func (j *MongoDBBackupJob) startBackup(ctx context.Context) error {
+func (j *MongoDBBackupJob) startBackup(ctx context.Context) (pbmBackup, error) {
```

Comment:
maybe return pointer in order not to create an empty struct in case of error ?

Reply:
It's micro optimization, but sure.
Allocating on heap, dereferencing ... can be slower then returning non-pointer.
Eg. https://medium.com/@vCabbage/go-are-pointers-a-performance-optimization-a95840d3ef85

Reply:
I have real examples where passing structs by value damaged performance dramatically (It was big structs and frequently invoked functions). 
We can refer here to the official review recommendations:
https://github.com/golang/go/wiki/CodeReviewComments#pass-values
https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

So the main idea is to pass by value something small and pass by pointer something big or something that may become big. Also if local variable is big enough it can be allocated on the heap even if it hasn't references at all: https://golang.org/doc/faq#stack_or_heap

In this particular situation, we most likely don't care, but a pointer is more common for our codebase.